pandas/tests/resample/test_base.py

```diff
@@ -118,6 +118,20 @@ def test_resample_empty_series(freq, empty_series, resample_method):
     assert_series_equal(result, expected, check_dtype=False)
 
 
+@pytest.mark.parametrize("freq", ["M", "D", "H"])
+@pytest.mark.parametrize("resample_method", ["count", "size"])
+def test_resample_count_empty_series(freq, resample_method):
+    # GH28427
+
+    empty_series = pd.Series([], dtype="datetime64[ns]", index=pd.DatetimeIndex([]))
+    result = getattr(empty_series.resample(freq), resample_method)()
+
+    expected = pd.Series([], dtype="int64", index=pd.DatetimeIndex([]))
+    expected.index = empty_series.index._shallow_copy(freq=freq)
```

Comment:
Ideally we wouldn't use private methods in tests, unless we're testing the private method. You can pass `freq` to the `DatetimeIndex` constructor, does that work?

Reply:
Yes, this works. It was taken from `test_resample_empty_series()` but it's not needed in these cases.